doc/guide/sample.hpp

```diff
@@ -16,6 +16,72 @@ e.g., @c make @c mlpack_knn or @c make @c mlpack_test; see @ref build.  However,
 any program that uses mlpack (and is not a part of the library itself) can be
 compiled easily with g++ or clang from the command line.
 
+@section compile_link_and_run Compile, Link and Run C++ Files 
+
+Typically the following commands would be used to compile and run the C++ files.
+
+@subsection compile_using_gcc Compile Using GCC
+
+@code{.sh}
+g++ -std=c++11 filename.cpp -o filename -lmlpack -larmadillo
+./filename
+@endcode
+
+If @c pkg-config is installed:
+
+@code{.sh}
+g++ -std=c++11 filename.cpp -o filename \
+`pkg-config --cflags --libs mlpack armadillo`
+./filename
+@endcode
+
+@subsection compile_using_clang Compile Using Clang
+
+@code{.sh}
+clang++ -Wall filename.cpp -o filename -lmlpack -larmadillo
+./filename
+@endcode
+
+If @c pkg-config is installed:
+
+@code{.sh}
+clang++ -Wall filename.cpp -o filename \
+`pkg-config --cflags --libs mlpack armadillo`
+./filename
+@endcode
+
+@note
+Flags like @c -lboost_serialization might be required depending upon the 
+code. If you are using LAPACK and BLAS instead of Armadillo the @c -llapack
+and @c -lblas would be required instead of @c -larmadillo
+
+@subsection compile_using_cmake Compile Using Cmake
+
+In /src
+  - filename.cpp
+  - CMakeLists.txt
+
+@code
+# CMakeLists.txt
+cmake_minimum_required(VERSION 3.10)
+
+# set the project name
+project(Project_Name VERSION 1.0)
+
+add_executable(Project_Name filename.cpp)
+
+target_link_libraries(Project_Name -lmlpack)
```

Comment:
@zoq Thanks for those links. I think:
1. If the code in the links you mentioned, would simply work via copy-paste (append) to the CMakeLists.txt, then we could just provide these links in our documentation.
2. If 1 doesn't work, creating a minimal example would be the way to go. 

Let me know 1 or 2?

If 2:

- 2.1 Some pointers on the minimal example would be helpful.
- 2.2 Should it be within this PR ?

Reply:
Let's go with option number 1, as I think it is less error prone.

Reply:
We probably don't need to copy in the contents of the find-modules, rather use find_package in the tutorial and link to the find-modules.
As another note, it's probably nice to show good practice and to use the variable names ARMADILLO_LIBRARIES  and MLPACK_LIBRARIES instead of -larmadillo and -lmlpack.

Reply:
@zoq @birm 
Thank you for your reviews. Really helpful. I've updated the documentation. 
Please review it as you get time.
:)